Lib/test/test_genericalias.py

```diff
@@ -65,6 +65,14 @@ class MyList(list):
         self.assertIs(t.__origin__, MyList)
         self.assertEqual(t.__parameters__, (int,))
 
+    def test_repr(self):
+        class MyList(list):
+            pass
+        self.assertEqual(repr(list[str]), 'list[str]')
+        self.assertEqual(repr(list[()]), 'list[()]')
+        self.assertEqual(repr(MyList[int]), 'test.test_genericalias.BaseTest.test_repr.<locals>.MyList[int]')
```

Comment:
If you run the test as a main script (as I do) the repr is different, `'__main__.BaseTest.test_repr.<locals>.MyList[int]'` -- so maybe do a .endswith() check.